### PR TITLE
[reactController] remove motionPlan, read trajectories directly from port

### PR DESCRIPTION
**Pull request by towardthesea**
The main purpose of this `PR` is:
- Remove `motionPlan` dependency 
- Introduce 1 new method to read trajectories from `planner` directly from port.

It is tested on simulation. The behavior of integration is the same!

cc @matejhof @pattacini

### CODE REVIEW

**Comment by alecive** [comment]
I cannot say anything about the implementation, but big :+1: about the design decision of decoupling the two parts of the software.

File: modules/reactController/src/reactCtrlThread.cpp [hunk 1]
```diff
@@ -408,76 +406,21 @@ void reactCtrlThread::run()
...
-                                additionalControlPointsVector.push_back(*controlPoint);
-                            }
-                        }
-                        else
-                            yWarning("[reactCtrlThread::run()] %d waypoints for control point %s < 1 - ignoring.",(*it).getNbWaypoint(),(*it).getCtrlPointName().c_str());
-                    
-                    }
-                    else //((*it).numberDimension < 3)
-                        yWarning("[reactCtrlThread::run()] %d dimensions specified for control point %s < 3 - ignoring.",(*it).getDimension(),(*it).getCtrlPointName().c_str());
+                    ControlPoint *ctrlPt = new ControlPoint();
```
Comment (pattacini) [inline modules/reactController/src/reactCtrlThread.cpp]: @towardthesea where is this dynamic object `ctrlPt` finally deallocated?

**Review by pattacini — CHANGES_REQUESTED** [review]
Aside from one doubt on dynamic allocation of resources, I've noticed that you changed the reading paradigm from **callback** to **polling**. Is that wanted / correct?

File: modules/reactController/src/reactCtrlThread.cpp [hunk 2]
```diff
@@ -419,6 +419,7 @@ void reactCtrlThread::run()
                     ctrlPt->type = "Elbow";
                     ctrlPt->x_desired = x_planned[1];
                     additionalControlPointsVector.push_back(*ctrlPt);
+                    delete ctrlPt;
```
Comment (pattacini) [inline modules/reactController/src/reactCtrlThread.cpp]: That's wrong...

**Review by pattacini — APPROVED** [review]

**Comment by towardthesea** [comment]
Regarding the **callback** vs **polling** question, `motionPlan` was designed with callback for the communication between `planner` and `supervisor`, which is asynchronous. In case of `react-controller`, the reference update is synchronous.

**Pull request merged**